pkg/controller/sync/controller.go

```diff
@@ -405,19 +416,20 @@ func (s *KubeFedSyncController) setFederatedStatus(fedResource FederatedResource
 	// If the underlying resource has changed, attempt to retrieve and
 	// update it repeatedly.
 	err := wait.PollImmediate(1*time.Second, 5*time.Second, func() (bool, error) {
-		if updateRequired, err := status.SetFederatedStatus(obj, reason, *collectedStatus); err != nil {
+		if updateRequired, err := status.SetFederatedStatus(obj, reason, *collectedStatus, *collectedResourceStatus, resourceStatusCollection); err != nil {
+			klog.Infof("Failed to set the status for %s %q", kind, name)
 			return false, errors.Wrapf(err, "failed to set the status")
 		} else if !updateRequired {
-			klog.V(4).Infof("No status update necessary for %s %q", kind, name)
+			klog.Infof("No status update necessary for %s %q", kind, name)
 			return true, nil
 		}
-
+		klog.Infof("Updating status of name '%v' and kind '%v'", name, kind)
```

Comment:
I'd suggest aligning the order of name/kind so it's always the same. Same goes for the type, either `%s` or `%v`.